src/irmin-pack/unix/file_manager.ml

```diff
@@ -786,6 +835,11 @@ struct
         pl.chunk_start_idx pl.chunk_num];
     Control.set_payload t.control pl
 
+  let add_volume t =
+    match t.lower with
+    | None -> Ok () (* TODO: throw exception? *)
```

Comment:
Any opinions on this?

Reply:
I think it's ok not throw an exception and just continue irmin-pack'ing along.

I'd suggest adding a docstring in the `File_manager` interface specifying that if there is no lower setup than the `add_volume` call is a nop.

Reply:
Maybe also add a log message indicating that `add_volume` was called without a lower being conigured.

Reply:
> I'd suggest adding a docstring in the `File_manager` interface specifying that if there is no lower setup than the `add_volume` call is a nop.

Correction: this should go in the `store_intf.mli`.

Reply:
Thanks for weighing in!

Usually I would agree that making this a no-op would be okay, but after thinking about this some more, I think it might be best to return an error to provide a clear signal to a client that something is likely wrong. It seems not ideal if a client calls this and thinks the operation succeeds but in reality they have not configured their store incorrectly. This would also be symmetrical with a call to GC failing when it is given the `Archive` action but no lower exists (not implemented yet but is the expected outcome).

I'll leave this as-is for the PR but follow up in a subsequent PR.

Reply:
I agree, especially since the function can already return an `Error`!